File: AIResources/Java/Direction.java
import java.util.concurrent.ThreadLocalRandom;

public class Direction
{
    public static final byte STILL = 0;
    public static final byte NORTH = 1;
    public static final byte EAST = 2;
    public static final byte SOUTH = 3;
    public static final byte WEST = 4;
    
    public static byte randomDirection() {
        return (byte)ThreadLocalRandom.current().nextInt(0, 5);
    }
}

File: airesources/Scala/Direction.scala
import scala.util.Random

object Direction {
  private val random = new Random()
  val CARDINALS = Seq(North, East, South, West)
  val ALL = Seq(Still, North, East, South, West)

  def getRandomDir: Direction = ALL(random.nextInt(5))
  def getRandomCard: Direction = CARDINALS(random.nextInt(4))
}

class Direction(value: Int) {
  def getValue = value
}

case object Still extends Direction(0)
case object North extends Direction(1)
case object East extends Direction(2)
case object South extends Direction(3)
case object West extends Direction(4)
